### PR TITLE
Move TwilioInteractor into own class

### PR DESCRIPTION
**Pull request by wwahammy**
- **Add initial TwilioRequestValidator**
- **Reorganize TwilioRequestValidator**
- **Rename TwilioRequestValidator -> TwilioInteractor**

### CODE REVIEW

File: lib/twilio-request-validator.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    )
+  end
+
+  def delete_entity(identity)
+    @verify_client.delete_entity(identity) unless identity.blank?
+  rescue StandardError => e
+    # 20404 means the resource does not exist. This can happen if an old unverified factor has
+    # already been cleaned up (i.e. deleted) by Verify.
+    raise e unless e.message.include?('20404')
+  end
```
Comment (wwahammy) [inline lib/twilio-request-validator.rb:26]: @BintLopez @camckin10 So, this isn't validation like the rest of the methods. So I think we should rename the class but not sure what?

File: lib/twilio-interactor.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
+class TwilioInteractor
```
Comment (BintLopez) [inline lib/twilio-interactor.rb:1]: I think we need test coverage here. Did we have test coverage for those private methods anywhere else?

File: app/controllers/devise/devise_authy_controller.rb [hunk 3]
```diff
@@ -23,6 +23,9 @@ class Devise::DeviseAuthyController < DeviseController
 
   before_action :initialize_twilio_verify_client
 
+  attr_reader :twilio_interactor
+  delegate :delete_entity, :registration_token_valid?, :register_totp, to: :twilio_interactor, private: true
```
Comment (BintLopez) [inline app/controllers/devise/devise_authy_controller.rb:27]: I dig the interactor refactor

File: lib/twilio-interactor.rb (hunk 2 above)
Comment (wwahammy) [reply lib/twilio-interactor.rb:1]: We didn't directly and I didn't see anything in Cashbox. We can add them though.

Comment (wwahammy) [reply lib/twilio-interactor.rb:1]: I've handled this!

**Comment by wwahammy** [comment]
@BintLopez I've added specs for TwilioInteractor, I think it's ready for final review.

**Review by camckin10 — APPROVED** [review]
Nice specs!  👍

**Review by BintLopez — APPROVED** [review]
🦑

**Pull request merged**